clientv1_test.go

```diff
@@ -0,0 +1,64 @@
+package sams
+
+import (
+	"testing"
+
+	"connectrpc.com/connect"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/sourcegraph/lib/pointers"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	clientsv1 "github.com/sourcegraph/sourcegraph-accounts-sdk-go/clients/v1"
+)
+
+func TestParseResponseAndError(t *testing.T) {
+	tests := []struct {
+		name    string
+		err     func() error
+		wantErr string
+	}{
+		{
+			name: "no error",
+			err: func() error {
+				return nil
+			},
+			wantErr: "",
+		},
+		{
+			name: "not found",
+			err: func() error {
+				return connect.NewError(connect.CodeNotFound, nil)
+			},
+			wantErr: ErrNotFound.Error(),
+		},
+		{
+			name: "record mismatch",
+			err: func() error {
+				detail, err := connect.NewErrorDetail(&clientsv1.ErrorRecordMismatch{})
+				require.NoError(t, err)
+				grpcErr := connect.NewError(connect.CodeUnknown, nil)
```

Comment:
Should we use the precondition failed code in these tests for presumably more realistic response?

Reply:
I am kinda debating hard in my head, on the one hand, precondition failed code is right for this particular error, but on the other hand it makes this generic error converter unnecessarily(?) complicated to digest. 🤔 

On the server-side tho, does return precondition failed code (in another PR)